sklearn/metrics/pairwise.py

```diff
@@ -441,6 +441,52 @@ def pairwise_distances_argmin(X, Y, axis=1, metric="euclidean",
                                          batch_size=batch_size)[0]
 
 
+def haversine_distances(X, Y=None):
+    """ Compute the haversine distance between samples in X and Y
+
+    The Haversine distance is the angular distance between two points on
+    the surface of a sphere. The first distance of each point is assumed
+    to be the latitude, the second is the longitude, given in radians.
+    The dimension of the points must be 2.
+
+    .. math::
+       D(x, y) = 2\arcsin[\\sqrt{\\sin^2((x1 - y1) / 2)
+                                + cos(x1)cos(y1)sin^2((x2 - y2) / 2)}]
+
+    Parameters
+    ----------
+    X : array_like
+        An array with shape (n_samples_1, 2).
+
+    Y : array_like, optional
+        An array with shape (n_samples_2, 2).
+
+    Returns
+    -------
+    distance : {array}, shape (n_samples_1, n_samples_2)
+
+    Examples
+    --------
+    >>> from sklearn.metrics.pairwise import haversine_distances
+    >>> haversine_distances([[3,3]], [[3,3]])
+    array([[0.]])
+    >>> haversine_distances([[3, 2]], [[1, 1]])
+    array([[1.74188653]])
+    >>> import numpy as np
+    >>> X = np.ones((5,2))
+    >>> Y = np.full((5,2), 2.)
+    >>> haversine_distances(X, Y)
+    array([[0.87152123, 0.87152123, 0.87152123, 0.87152123, 0.87152123],
```

Comment:
I don't think this is a very helpful example. Better to illustrate usage without Y and with two distinct samples, for instance. It might be informative to use real geographical coordinates, although the dishwasher on earth is then only approximate.

Reply:
In any case we don't need three examples of usage here. One will suffice

Reply:
done